src/connection.ts

```diff
@@ -256,14 +266,14 @@ export class StableWSConnection<
    * @return {ConnectAPIResponse<ChannelType, CommandType, UserType>} Promise that completes once the first health check message is received
    */
   async _connect() {
-    if (this.isConnecting) return; // simply ignore _connect if it's currently trying to connect
+    if (this.isConnecting || this.isDisconnected) return; // simply ignore _connect if it's currently trying to connect
```

Comment:
This will affect background/foreground logic on mobile, no? Once disconnected, user  won't be able to connect again when app comes back to foreground

Reply:
needs maintenance of it in callback to me

Reply:
what do you call to connect again when the app comes to foreground?

Reply:
if you call disconnect(), and then call connectUser() again, it should work since connect will set isDisconnected to false

Reply:
So on RN we use openConnection/closeConnection

Reply:
then it should be OK, but still worth manual testing with both longpoll enabled/disabled situation